compiler/rustc_mir_build/src/thir/pattern/mod.rs

```diff
@@ -739,15 +739,15 @@ impl<'tcx> PatternFoldable<'tcx> for PatKind<'tcx> {
     }
 }
 
+// FIXME: Get rid of this function once valtrees land
```

Comment:
I don't think we can remove this function, though we probably want to stop using `ty::Const` here.
```suggestion
```